src/util/query_exec_util.cpp

```diff
@@ -0,0 +1,249 @@
+    return std::make_pair(std::move(statement), nullptr);
+  }
+
+  try {
+    // TODO(wz2): Specify params?
+    auto binder = binder::BindNodeVisitor(common::ManagedPointer(accessor), db_oid_);
+    binder.BindNameToNode(statement->ParseResult(), params, param_types);
+  } catch (std::exception &e) {
+    // Caught a binding exception
+    return {nullptr, nullptr};
+  }
+
+  auto out_plan = trafficcop::TrafficCopUtil::Optimize(txn, common::ManagedPointer(accessor), statement->ParseResult(),
+                                                       db_oid_, stats_, std::move(cost), optimizer_timeout_)
+                      ->TakePlanNodeOwnership();
+  return std::make_pair(std::move(statement), std::move(out_plan));
+}
+
+bool QueryExecUtil::ExecuteDDL(const std::string &query) {
+  NOISEPAGE_ASSERT(txn_ != nullptr, "Requires BeginTransaction() or UseTransaction()");
+  auto txn = common::ManagedPointer<transaction::TransactionContext>(txn_);
+  auto accessor = catalog_->GetAccessor(txn, db_oid_, DISABLED);
+  auto result = PlanStatement(query, nullptr, nullptr, std::make_unique<optimizer::TrivialCostModel>());
+  const std::unique_ptr<network::Statement> &statement = result.first;
+  const std::unique_ptr<planner::AbstractPlanNode> &out_plan = result.second;
+  NOISEPAGE_ASSERT(!network::NetworkUtil::DMLQueryType(statement->GetQueryType()), "ExecuteDDL expects DDL statement");
+
+  // Handle SET queries
+  bool status = true;
+  if (statement->GetQueryType() == network::QueryType::QUERY_SET) {
+    const auto &set_stmt = statement->RootStatement().CastManagedPointerTo<parser::VariableSetStatement>();
+    settings_->SetParameter(set_stmt->GetParameterName(), set_stmt->GetValues());
+    status = true;
+  } else {
+    switch (statement->GetQueryType()) {
+      case network::QueryType::QUERY_CREATE_TABLE:
+        status = execution::sql::DDLExecutors::CreateTableExecutor(
+            common::ManagedPointer<planner::CreateTablePlanNode>(
+                reinterpret_cast<planner::CreateTablePlanNode *>(out_plan.get())),
+            common::ManagedPointer(accessor), db_oid_);
+        break;
+      case network::QueryType::QUERY_DROP_INDEX:
+        // Drop index does not need execution of compiled query
+        status =
+            execution::sql::DDLExecutors::DropIndexExecutor(common::ManagedPointer<planner::AbstractPlanNode>(out_plan)
+                                                                .CastManagedPointerTo<planner::DropIndexPlanNode>(),
+                                                            common::ManagedPointer<catalog::CatalogAccessor>(accessor));
+        break;
+      case network::QueryType::QUERY_CREATE_INDEX:
+        // TODO(lin): We actually don't need to populate the index tuples after creating the index placeholder for the
+        //  "what-if" API. But since we need to execute the query to get the features, we need to compile and execute
+        //  the query for now.
```

Comment:
We actually do need to compile and execute `CREATE_INDEX` for now. See the implementation in the original `pilot_util.cpp`.

Reply:
Unified the strangely placed code inside of `pilot_util.cpp`.